packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -524,7 +524,13 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
       />
       {cells}
       {emptyCellWidth > 0 && <EmptyCell width={emptyCellWidth} />}
-      {rightCells.length > 0 && <div role="presentation" className={gridClasses.filler} />}
+      {rightCells.length > 0 && (
+        <div
+          role="presentation"
+          className={clsx(gridClasses.filler, gridClasses['filler--borderTop'])}
```

Comment:
Is this space being filled by the filler:
https://github.com/mui/mui-x/assets/12609561/33556350-6f25-40d5-baa3-287fbd54408c

Probably, need to adjust it during scroll.

Reply:
Could you give me more details about how to reproduce that? It's working fine on my end.

Reply:
I can observe this small glitch while resizing the right pinned column on every browser (Chrome, Firefox, Edge, Safari) in Mac OS.
The trick to reproduce is to first scroll to the right-most pixel of the scrollable area (for non-pinned columns) and then try to resize the right-pinned column without stopping the mouse down. It gets fixed on mouse up though. Let me know if you can reproduce it in Linux or guess the possible reason. I'll take it otherwise.

Reply:
If you can take a look that would be very nice, I can't reproduce on linux:

[Screencast from 2024-05-15 23-54-04.webm](https://github.com/mui/mui-x/assets/1423607/b6c75d4d-0481-4796-b5d3-3ba7949d1157)

Reply:
I have finished handling this case. It feels like the column resize code is becoming more ugly every time I touch it.

Reply:
I'm not sure if it's the same issue, but I can still reproduce some weird behavior in https://codesandbox.io/p/sandbox/flamboyant-wing-wdkzzl (it uses the latest package builds from this branch):


https://github.com/mui/mui-x/assets/13808724/1dc82118-315b-4d45-b611-5d50aa3a2d05

Reply:
Yes, it still causes issues.

A dumb way to solve it during my debugging was to call `apiRef.current.setColumnWidth` behind a debounce (which actually updates the state and causes recomputation) in `updateWidth`, which fixed it with visible flickering. I am not sure if it could be done while not hurting the UX (performance-wise) but one idea could be to involve some state updation for the specific column the user is resizing for proper re-computation to happen.

Reply:
I've been thinking about doing state updates as well. The complexity of the resizing code is getting very high with this PR, and I feel like I'm playing whack-a-mole with edge cases.

Reply:
I cannot reproduce the issue I posted in this thread with `disableVirtualization={true}`, so it seems like the virtualization still kicks in, even after https://github.com/mui/mui-x/pull/12979#discussion_r1620162933